### ISSUE TITLE
Debugging of python code without saving to file

### ISSUE THREAD

**Issue by DonJayamanne** [Microsoft/ptvsd#704]
Add support for debugging of python scripts/code that has not been persisted to a file.
Similar to `python -c`.
If i'm not mistaken PTVSD 3.x supported this, hence VS might need this feature.

**Comment by DonJayamanne** [comment]
The old version of PTVSD did support this.
Here's the original code https://github.com/Microsoft/vscode-python/blob/master/pythonFiles/PythonTools/visualstudio_py_launcher.py#L67

https://github.com/Microsoft/vscode-python/blob/master/pythonFiles/PythonTools/ptvsd/visualstudio_py_debugger.py#L2621

**Comment by DonJayamanne** [comment]
@huguesv 
Is it possible in PTVS to debug python code without saving to file?

**Comment by huguesv** [comment]
I don't think so. Even when you create a new file in the editor without saving, the file exists in the temp folder. We prompt to save and run the empty temp file, if the user didn't choose to save.

I think this must be only for when the user puts a `-c` in the launch arguments (instead of specifying a startup file).

**Comment by DonJayamanne** [comment]
> I think this must be only for when the user puts a -c

Is this supported in VS? If it is, then this feature will be required for parity.

**Comment by huguesv** [comment]
In VS 2017, some users reported being confused when launching with no startup file (nothing happened), so we added a check for a startup file, not realizing we were now preventing launching with `-m` and `-c`.

It will work again with https://github.com/Microsoft/PTVS/pull/4601

**Comment by DonJayamanne** [comment]
If that's the case, this is not an enhancement, but required, for parity.

**Comment by karthiknadig** [comment]
@int19h now that we have `-c` this should be supported right?

**Comment by int19h** [comment]
Not quite. You can pass code to `-c`, but you wouldn't be able to set breakpoints in it (since it doesn't have a filename for a breakpoint to match). And if you pause, the IDE won't be able to show the source code, since it doesn't know where to get that from. 

Now, if that code runs something that *is* in a file, then you can debug that file normally. Or if you manually specify the filename when invoking `compile()` before `exec()`, and that filename corresponds to something on disk... but neither of those are directly related to `-c`.

These are all the same restrictions as 3.x had tho.
